src/compiler/checker.ts

```diff
@@ -13082,18 +13082,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
         }
         setStructuredTypeMembers(type, members, emptyArray, emptyArray, indexInfos || emptyArray);
+
+        const exportSymbol = getExportSymbolOfValueSymbolIfExported(symbol);
+
         // We resolve the members before computing the signatures because a signature may use
         // typeof with a qualified name expression that circularly references the type we are
         // in the process of resolving (see issue #6072). The temporarily empty signature list
         // will never be observed because a qualified name can't reference signatures.
-        if (symbol.flags & (SymbolFlags.Function | SymbolFlags.Method)) {
+        if (exportSymbol.flags & (SymbolFlags.Function | SymbolFlags.Method)) {
             type.callSignatures = getSignaturesOfSymbol(symbol);

```

Comment:
it's worth noting here that signatures here are still obtained from the `symbol` and not from `exportSymbol`, using `exportSymbol` breaks function overloads displays in [`functionOverloadErrors`](https://github.com/microsoft/TypeScript/blob/996aeb6aaf8d6a04f5b9f29085f062bca98642b2/tests/cases/conformance/functions/functionOverloadErrors.ts) and [`overloadModifiersMustAgree`](https://github.com/microsoft/TypeScript/blob/996aeb6aaf8d6a04f5b9f29085f062bca98642b2/tests/cases/compiler/overloadModifiersMustAgree.ts)

Reply:
What breaks in those tests if you get signatures from the export symbol?

Reply:
```diff
diff --git a/src/compiler/checker.ts b/src/compiler/checker.ts
index 4b556122f5..510eb6a794 100644
--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -13086,7 +13086,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         // in the process of resolving (see issue #6072). The temporarily empty signature list
         // will never be observed because a qualified name can't reference signatures.
         if (exportSymbol.flags & (SymbolFlags.Function | SymbolFlags.Method)) {
-            type.callSignatures = getSignaturesOfSymbol(symbol);
+            type.callSignatures = getSignaturesOfSymbol(exportSymbol);
         }
         // And likewise for construct signatures for classes
         if (exportSymbol.flags & SymbolFlags.Class) {
diff --git a/tests/baselines/reference/functionOverloadErrors.types b/tests/baselines/reference/functionOverloadErrors.types
index a0b1ceb121..dcb12a43e0 100644
--- a/tests/baselines/reference/functionOverloadErrors.types
+++ b/tests/baselines/reference/functionOverloadErrors.types
@@ -168,14 +168,14 @@ module M {
 >fn1 : () => any
 
     function fn1(n: string);
->fn1 : { (): any; (n: string): any; }
+>fn1 : () => any
 >n : string
 
     function fn1() { } 
->fn1 : { (): any; (n: string): any; }
+>fn1 : () => any
 
     function fn2(n: string);
->fn2 : { (n: string): any; (): any; }
+>fn2 : () => any
 >n : string
 
     export function fn2();
diff --git a/tests/baselines/reference/overloadModifiersMustAgree.types b/tests/baselines/reference/overloadModifiersMustAgree.types
index f3c2665bef..bd3ff1df34 100644
--- a/tests/baselines/reference/overloadModifiersMustAgree.types
+++ b/tests/baselines/reference/overloadModifiersMustAgree.types
@@ -13,14 +13,14 @@ class baz {
 }
 
 declare function bar();
->bar : { (): any; (s: string): any; }
+>bar : (s: string) => any
 
 export function bar(s: string);
 >bar : (s: string) => any
 >s : string
 
 function bar(s?: string) { }
->bar : { (): any; (s: string): any; }
+>bar : (s: string) => any
 >s : string
 
 interface I {
```